New Visualization Navigation

achang28 opened the pull request:
As usual, not yet ready to merge.

achang28 commented:
@jingchunzhu @maryjgoldman pull from my repo and checkout the KM Plot button (https://github.com/achang28/ucsc-xena-client/tree/NewNavigation).  Still a big Work in Progress, but at the very minimum the KM Plot graph is showing up...

jingchunzhu commented:
I clicked on KM Plot, but no KM plot showing up. This is on TCGA Breast cancer cohort.

jingchunzhu commented:
The new black navbar really brothers me, I like our current production's "non-inverse" (or default)  look and feel.   Attached a diff to use what's on production look.
[nav.js.diff.txt](https://github.com/ucscXena/ucsc-xena-client/files/205903/nav.js.diff.txt)

The diff requires a js/nav.css file, contents below

.navbar .nav li a:hover
{
    background-color:#e7e7e7
}

.navbar-default .navbar-nav > .active > a, .navbar-default .navbar-nav > .active > a:hover, .navbar-default .navbar-nav > .active > a:focus
{
    color:#ff6600;
    background-color:white
}

achang28 commented on js/ChartView.js:
```diff
@@ -2,9 +2,12 @@
+		if (!_.isEqual(newProps.appState, this.props.appState))
+			debugger;
 		return false;
 	},
```
@acthp Do you know exactly what the application is considering when determining whether to render the actual chart, or the appropriate error message?

From the img below, while on the Chart view, switching cohort does not dismiss the previous chart view.  I went from 'TCGA PANCAN12' to 'ccle'.  The master branch currently behaves this same way as well (I turned off the disable parameter on the Cohort drop-down menu in order to test this issue out).  From the above code changes, I was trying to see what differences to consider between new appState vs. old appState.

<img width="784" alt="incorrect render" src="https://cloud.githubusercontent.com/assets/3760499/14343811/3362bf46-fc58-11e5-8051-92bbdb584902.png">

achang28 commented:
@jingchunzhu Pull the latest codebase.  I incorporated the latest updates from the Master branch in production, and also incorporated the subtle visual changes you mentioned 2 days ago.

jingchunzhu commented:
good work-in-progress.

Jing

On Thu, Apr 7, 2016 at 6:34 PM, Albert Chang <notifications@github.com>
wrote:

> @jingchunzhu <https://github.com/jingchunzhu> Pull the latest codebase. I
> incorporated the latest updates from the Master branch in production, and
> also incorporated the subtle visual changes you mentioned 2 days ago.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/ucscXena/ucsc-xena-client/pull/83#issuecomment-207163497>
>

achang28 commented on js/controllers/common.js:
```diff
@@ -63,6 +63,7 @@ var setCohort = (state, cohort) =>
 				"columnOrder", [],
 				"data", {},
 				"datasets", null,
+				"mode", 'heatmap',
```
@acthp Rather than hard-coding the value, where would I source the default ```mode``` from?
